src/particle.cpp

```diff
@@ -6,14 +6,13 @@
 #include <string>
 
 // constructor
-sample_particle::sample_particle(std::string filename)
+sample_particle::sample_particle(std::string filename, int size)
 {
   set_filename(filename);
 
   open_file();
   
-  //read_file(filename);
-  
+  set_batch_size(size); 
```

Comment:
could you move this upto line 11, perhaps add a comment to describe why we set batch size?